### PR TITLE
feat(metrics): add phase metrics for vm,disk,vmbda

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
add 3 metrics:
virtualization_virtualmachine_status_phase
virtualization_virtualmachineblockdeviceattachment_status_phase
virtualization_virtualdisk_status_phase

## Why do we need it, and what problem does it solve?
We will be able to monitor resource changes

## Checklist
- [ ] The code is covered by unit tests.
- [X] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/virtualization-artifact/pkg/controller/vmbda_controller.go [hunk 1]
```diff
@@ -48,7 +52,21 @@ func NewVMBDAController(
...
+	log.Info("Initialized VirtualMachineBlockDeviceAttachment controller")
 	return cvmiController, nil
 }
+
+type vmbdaLister struct {
+	vmbdaCache cache.Cache
+}
+
+func (l vmbdaLister) List() ([]v1alpha2.VirtualMachineBlockDeviceAttachment, error) {
+	vmdas := v1alpha2.VirtualMachineBlockDeviceAttachmentList{}
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/vmbda_controller.go:66]: vmdas -> vmbdas

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+func (c Collector) Describe(ch chan<- *prometheus.Desc) {
+	for _, v := range vmbdaMetrics {
+		ch <- v
+	}
+}
+
+func (c Collector) Collect(ch chan<- prometheus.Metric) {
+	vmbdas, err := c.lister.List()
+	if len(vmbdas) == 0 || err != nil {
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:48]: Error handling is lost

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func (s *scraper) updateVMBDAStatusPhaseMetrics(vmbda virtv2.VirtualMachineBlockDeviceAttachment) {
+	phase := vmbda.Status.Phase
+	if phase == "" {
+		phase = virtv2.BlockDeviceAttachmentPhaseInProgress
+	}
+	phases := []struct {
+		v bool
+		n string
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:76]: v - value? 
what is n?

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	}{
+		{phase == virtv2.BlockDeviceAttachmentPhaseInProgress, string(virtv2.BlockDeviceAttachmentPhaseInProgress)},
+		{phase == virtv2.BlockDeviceAttachmentPhaseAttached, string(virtv2.BlockDeviceAttachmentPhaseAttached)},
+		{phase == virtv2.BlockDeviceAttachmentPhaseFailed, string(virtv2.BlockDeviceAttachmentPhaseFailed)},
+	}
+	desc := vmbdaMetrics[MetricVMBDAStatusPhase]
+	for _, p := range phases {
+		mv, err := prometheus.NewConstMetric(
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:84]: What is mv?

**Review by Isteb4k — APPROVED** [review]

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go (hunk 4 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:84]: rename to metric

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go (hunk 2 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:48]: added logging

File: images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go (hunk 3 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/pkg/monitoring/metrics/vmbda/collector.go:76]: n - name.
renamed

**Pull request merged**